### ISSUE TITLE
Please document random and deepseq

### ISSUE THREAD

**Issue by ramsdell** [GaloisInc/cryptol#1001]
I notice that random and deepseq have been available in the interpreter at least since Cryptol version 2.8.0.  I have not been able to find a description of these two functions.  If I haven't missed something, please add words that describe these two functions.

**Comment by robdockins** [comment]
You can get documentation on these functions directly from the REPL.

```
Cryptol> :help random

    random : {a} [256] -> a

Generates random values from a seed.  When called with a function, currently
generates a function that always returns zero.

Cryptol> :help deepseq

    deepseq : {a, b} (Eq a) => a -> b -> b

A strictness-increasing operation.  The first operand
is reduced to normal form before evaluating the second
argument.

The Eq constraint restricts this operation to types
where reduction to normal form makes sense.
```

**Comment by ramsdell** [comment]
The :help feature is very useful, but the lack of any mention of these functions in documentation means I didn't know to ask the interpreter about them.  I only realized they were there by looking at share/doc/cryptol/CryptolPrims.pdf.  Perhaps there is a simple script one could write that takes a list of primitives and causes Cryptol to produce help on each one and collects the output into a text file or maybe an appendix for ProgrammingCryptol.pdf.

**Comment by robdockins** [comment]
Yes, finding a good way to autogenerate that kind of documentation would be a big win.  I think we're probably pretty close to being able to do something like that.

**Comment by brianhuffman** [comment]
See #616.

**Comment by ramsdell** [comment]
Here is a script that fills the gap until a more polished document generator solution becomes available.

[crydoc.sh.txt](https://github.com/GaloisInc/cryptol/files/5669224/crydoc.sh.txt)

**Comment by ramsdell** [comment]
There is a small bug in the script I sent.  The argument to sed should be $1, not "$1".  Sorry about that.

**Comment by ramsdell** [comment]
I discovered some symbols have more than one definition.  Here is a script that takes that fact into account.

[crydoc.sh.txt](https://github.com/GaloisInc/cryptol/files/5672922/crydoc.sh.txt)

**Comment by brianhuffman** [comment]
In order to close this ticket, we should have markdown or latex documentation auto-generated from the cryptol prelude, and included in the distribution.

**Comment by ramsdell** [comment]
If the text file that is generated by the script I submitted is included in the share/doc/cryptol directory of a binary release, I would consider this issue closed.  I discourage translating the text output into Markdown.  Who wants to start up a Markdown previewer to see something that can easily be read in your favorite editor?

**Comment by ramsdell** [comment]
What is bin/cryptol-html and cryptol-remote-api and where are they documented?  I found nothing about them in share/doc/cryptol.

**Comment by weaversa** [comment]
> If the text file that is generated by the script I submitted is included in the share/doc/cryptol directory of a binary release, I would consider this issue closed. I discourage translating the text output into Markdown. Who wants to start up a Markdown previewer to see something that can easily be read in your favorite editor?

I kinda like markdown. Both GitHub and VSCode (by some accounts the most widely used development platform) provide native markdown views.

**Comment by ramsdell** [comment]
Full disclosure, I tried to come up with a way to translate the text output into Markdown, but I couldn't think of a way that added value by doing so.

**Comment by yav** [comment]
`crytpol-html` converts Cryptol source to `html` by syntax-highlighted html using Cryptol's lexer.   The separate binary is mostly a left over from a long time ago, and was helpful while I was writing the lexer.   If this is useful functionality we should probably add it to `cryptol` proper

**Comment by robdockins** [comment]
Does the documentation provided by the `crydoc.sh` script above have any real advantages over just reading the original `Cryptol.cry` source file?  Those built-in modules aren't currently included into the source tarball, but I think they probably should be (CF #1074).

**Comment by ramsdell** [comment]
Ideally, both the output of crydoc.sh and Cryptol.cry should be included in a binary distribution, but as long as at least one of them are, I could have figured that random and deepseq are provided.  So if you include at least one of these two items in a binary distribution, we can close this issue.
